packages/iov-bns-governance/src/governor.ts

```diff
@@ -0,0 +1,109 @@
+import BN = require("bn.js");
+
+import { Address, Identity, WithCreator } from "@iov/bcp";
+import {
+  bnsCodec,
+  BnsConnection,
+  BnsTx,
+  CreateProposalTx,
+  Electorate,
+  Proposal,
+  TallyTx,
+  VoteOption,
+  VoteTx,
+} from "@iov/bns";
+import { Encoding } from "@iov/encoding";
+
+import { CommitteeId, ProposalOptions, ProposalType } from "./proposals";
+
+/* Constants to be finalised once the committees are actually on-chain */
+export const COMMITTEE_IDS = {
+  testnet: {
+    GOVERNING_BOARD: 1 as CommitteeId,
+    ECONOMIC_COMMITTEE: 2 as CommitteeId,
+    VALIDATOR_COMMITTEE: 3 as CommitteeId,
+    TECH_COMMITTEE: 4 as CommitteeId,
+  },
+};
+
+export interface GovernorOptions {
+  readonly connection: BnsConnection;
+  readonly identity: Identity;
+}
+
+export class Governor {
+  private readonly connection: BnsConnection;
+  private readonly identity: Identity;
+  private readonly address: Address;
+
+  public constructor({ connection, identity }: GovernorOptions) {
+    this.connection = connection;
+    this.identity = identity;
+    this.address = bnsCodec.identityToAddress(this.identity);
+  }
+
+  public async getElectorates(): Promise<readonly Electorate[]> {
+    const electorates = await this.connection.getElectorates();
+    return electorates.filter(({ electors }) => Object.keys(electors).some(key => key === this.address));
+  }
+
+  public async getElectionRuleId(electorateId: number): Promise<string> {
+    const electionRules = await this.connection.getElectionRules();
+    const filteredRules = electionRules.filter(rule => rule.electorateId === electorateId);
+
+    if (filteredRules.length === 0) {
+      throw new Error("No election rule found for electorate");
+    }
+
+    const mostRecentRule = filteredRules
+      .slice(1)
+      .reduce((rule1, rule2) => (rule1.version > rule2.version ? rule1 : rule2), filteredRules[0]);
+
+    return Encoding.toHex(mostRecentRule.id);
```

Comment:
Let's use `return new BN(mostRecentRule.id).toNumber()` here instead to hide the lack of beauty of @iov/bns

See https://github.com/iov-one/iov-core/issues/1110